steely/plugins/countdown/solver.py

```diff
@@ -0,0 +1,81 @@
+from collections import deque
+
+
+class TrieNode:
+
+    def __init__(self, key=None):
+        self.key = key
+        self.children = {}
+        self.is_word_end = False
+
+    def add_descendant(self, word):
+        if not len(word):
+            self.is_word_end = True
+            return
+
+        k = word[0]
+        if k not in self.children:
+            self.children[k] = TrieNode(key=k)
+        self.children[k].add_descendant(word[1:])
+
+
+class Trie:
+
+    def __init__(self):
+        self.root = TrieNode()
+
+    def add_word(self, word):
+        self.root.add_descendant(word)
+
+    def get_words_for_letters(self, letters):
+        words = set()
+        n = len(letters)
+        q = deque([(self.root, '', [False for _ in range(n)])])
```

Comment:
`[False] * n` might be nicer

Reply:
I tend to avoid multiplying []s, because it can sometimes be hard to figure out if it's multiple copies of one obj in a list or multiple separate lists. Here it works though, I'll make the change.